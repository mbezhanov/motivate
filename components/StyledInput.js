import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View } from 'react-native';
import Colors from '../constants/Colors';
import Sizes from '../constants/Sizes';

class StyledInput extends Component {

  state = { focused: false };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{this.props.label}</Text>
        <TextInput
          style={[styles.textInput, this.state.focused ? styles.focused : null]}
          underlineColorAndroid="transparent"
          onFocus={this._handleFocus}
          onBlur={this._handleBlur}
          multiline={this.props.multiline}
          value={this.props.value}
          onChangeText={this.props.onChangeText}
        />
      </View>
    );
  };

  _handleFocus = () => {
    this.setState({ focused: true });
  };

  _handleBlur = () => {
    this.setState({ focused: false });
  };
}

StyledInput.defaultProps = { multiline: false };

const styles = StyleSheet.create({
  container: {
    margin: Sizes.M,
  },
  label: {
    color: Colors.TEXT,
    fontSize: Sizes.S,
  },
  textInput: {
    color: Colors.TEXT,
    borderBottomWidth: 1,
    borderColor: Colors.BORDER,
    fontSize: Sizes.M,
    marginTop: Sizes.S,
  },
  focused: {
    borderColor: Colors.TEXT,
  }
});

export default StyledInput;
