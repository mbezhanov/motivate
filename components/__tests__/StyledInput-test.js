import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import StyledInput from '../StyledInput';
import { TextInput } from 'react-native';

const _focus = (component) => {
  const textInput = component.findByType(TextInput);
  textInput.props.onFocus();
};

const _blur = (component) => {
  const textInput = component.findByType(TextInput);
  textInput.props.onBlur();
};

describe('StyledInput', () => {
  let tree;

  beforeEach(() => {
    tree = renderer.create(<StyledInput />);
  });

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('starts having a highlight effect when focused', () => {
    const a = tree.toJSON();
    _focus(tree.root);
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });

  it('stops having a highlight effect when unfocused', () => {
    _focus(tree.root);
    const a = tree.toJSON();
    _blur(tree.root);
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });
});
