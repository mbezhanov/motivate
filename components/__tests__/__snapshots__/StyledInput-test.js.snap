// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#bbb",
        "fontSize": 12,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    multiline={false}
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#555",
          "color": "#bbb",
          "fontSize": 16,
          "marginTop": 12,
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`StyledInput starts having a highlight effect when focused 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,11 +25,13 @@
            \\"borderColor\\": \\"#555\\",
            \\"color\\": \\"#bbb\\",
            \\"fontSize\\": 16,
            \\"marginTop\\": 12,
          },
-         null,
+         Object {
+           \\"borderColor\\": \\"#bbb\\",
+         },
        ]
      }
      underlineColorAndroid=\\"transparent\\"
    />
  </View>"
`;

exports[`StyledInput stops having a highlight effect when unfocused 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -25,13 +25,11 @@
            \\"borderColor\\": \\"#555\\",
            \\"color\\": \\"#bbb\\",
            \\"fontSize\\": 16,
            \\"marginTop\\": 12,
          },
-         Object {
-           \\"borderColor\\": \\"#bbb\\",
-         },
+         null,
        ]
      }
      underlineColorAndroid=\\"transparent\\"
    />
  </View>"
`;
